horovod/torch/__init__.py

```diff
@@ -122,22 +126,71 @@ def DistributedOptimizer(optimizer, named_parameters=None):
     return cls(optimizer.param_groups, named_parameters)
 
 
+def broadcast_object(obj, root_rank, name=None):
+    """
+    Serializes and broadcasts an object from root rank to all other processes.
+    Typical usage is to broadcast the `optimizer.state_dict()`, for example:
+
+    ```
+    state_dict = broadcast_object(optimizer.state_dict(), 0)
+    if hvd.rank() > 0:
+        optimizer.load_state_dict(state_dict)
+    ```
+
+    Arguments:
+        obj: An object capable of being serialized without losing any context.
+        root_rank: The rank of the process from which parameters will be
+                   broadcasted to all other processes.
+        name: Optional name to use during broadcast, will default to the class
+              type.
+
+    Returns:
+        The object that was broadcast from the `root_rank`.
+    """
+    if name is None:
+        name = str(type(obj))
+
+    if rank() == root_rank:
+        b = io.BytesIO()
+        pickle.dump(obj, b)
+        buf = bytearray(b.getvalue())
+        t = torch.ByteTensor(buf)
+        sz = torch.IntTensor([t.shape[0]])
+        broadcast_(sz, root_rank, name + '.sz')
+    else:
+        sz = torch.IntTensor([0])
+        broadcast_(sz, root_rank, name + '.sz')
+        t = torch.ByteTensor([0] * sz.tolist()[0])
```

Comment:
What if the tensor is 5GB?  `*` may be quite inefficient.

Reply:
You can just specify the size, no need for zeroing out.

Reply:
Fixed.